auth_ldap_attribute_sync/models/res_company_ldap.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (C) 2018 Brainbean Apps
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class CompanyLDAP(models.Model):
+    _inherit = 'res.company.ldap'
+    _description = 'Company LDAP configuration'
+
+    name_attribute = fields.Char(string='Name attribute', default='cn')
+    name_attribute_mode = fields.Selection(
+        string='Name attribute sync mode',
+        selection='_attribute_modes',
+        default='initial'
+    )
+
+    email_attribute = fields.Char(string='Email attribute', default='mail')
+    email_attribute_mode = fields.Selection(
+        string='Email attribute sync mode',
+        selection='_attribute_modes',
+        default='initial'
+    )
+
+    phone_attribute = fields.Char(string='Phone attribute')
+    phone_attribute_mode = fields.Selection(
+        string='Phone attribute sync mode',
+        selection='_attribute_modes'
+    )
+
+    mobile_attribute = fields.Char(string='Mobile attribute')
+    mobile_attribute_mode = fields.Selection(
+        string='Mobile attribute sync mode',
+        selection='_attribute_modes'
+    )
+
+    function_attribute = fields.Char(string='Jon Position attribute')
+    function_attribute_mode = fields.Selection(
+        string='Jon Position attribute sync mode',
+        selection='_attribute_modes'
+    )
+
+    def _fields_to_sync(self):
+        return [
+            'name',
+            'email',
+            'phone',
+            'mobile',
+            'function'
+        ]
+
+    def _attribute_modes(self):
+        return [
+            ('initial', 'On user creation'),
+            ('always', 'On every login')
+        ]
+
+    def _get_ldap_dicts(self):
+        res = super()._get_ldap_dicts()
+        for rec in res:
+            ldap = self.sudo().browse(rec['id'])
+            for field in self._fields_to_sync():
+                rec['%s_attribute' % field] = getattr(
```

Comment:
no need for `getattr`, just use the [string index](https://github.com/OCA/OCB/blob/12.0/odoo/models.py#L5025) here too

Reply:
Done